task-maker-cache/src/lib.rs

```diff
@@ -105,10 +104,8 @@ const CACHE_FILE: &str = "cache.bin";
 /// Handle the cached executions, loading and storing them to disk.
 #[derive(Debug)]
 pub struct Cache {
-    /// Entries to flush to disk.
-    to_flush: HashMap<u8, CacheFile>,
-    /// The base directory where the cache files are stored.
-    cache_dir: PathBuf,
+    /// Cache entries.
+    file: CacheFile,
```

Comment:
I left the `CacheFile` struct in here to keep the file format and the cache implementation separated, even though we only have a single file. I don't really have a strong opinion on this though